org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/extensions/ErrorParticipantLanguageServiceTest.java

```diff
@@ -0,0 +1,504 @@
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.services.extensions;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lemminx.XMLAssert;
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.dom.DOMAttr;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.extensions.contentmodel.participants.XMLSyntaxErrorCode;
+import org.eclipse.lemminx.extensions.contentmodel.settings.ContentModelSettings;
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.eclipse.lemminx.services.DocumentSymbolsResult;
+import org.eclipse.lemminx.services.SymbolInformationResult;
+import org.eclipse.lemminx.services.XMLLanguageService;
+import org.eclipse.lemminx.services.extensions.codelens.ICodeLensParticipant;
+import org.eclipse.lemminx.services.extensions.codelens.ICodeLensRequest;
+import org.eclipse.lemminx.services.extensions.diagnostics.IDiagnosticsParticipant;
+import org.eclipse.lemminx.services.extensions.format.IFormatterParticipant;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.settings.XMLSymbolFilter;
+import org.eclipse.lemminx.settings.XMLSymbolSettings;
+import org.eclipse.lemminx.utils.XMLBuilder;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeLens;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.DocumentHighlight;
+import org.eclipse.lsp4j.DocumentLink;
+import org.eclipse.lsp4j.DocumentSymbol;
+import org.eclipse.lsp4j.Hover;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.LocationLink;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.ReferenceContext;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.SymbolKind;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests to ensure that if a participant throws an exception, the language
+ * server still functions
+ *
+ * @author datho7561
+ */
+public class ErrorParticipantLanguageServiceTest {
+
+	/**
+	 * Adds participants that throw runtime exceptions, as well as simple
+	 * participants that always return the same result. Does not include all the
+	 * default participants.
+	 */
+	static class ErrorParticipantLanguageService extends XMLLanguageService {
+
+		public static final CodeLens TEST_CODE_LENS = XMLAssert.cl(XMLAssert.r(0, 0, 0, 0), "a", "a");
+		public static final LocationLink TEST_LOCATION_LINK = XMLAssert.ll("file:///hello", XMLAssert.r(0, 0, 0, 2),
```

Comment:
This test which uses this constants fails on Windows OS, because file:///hello i snot a valid path. I suggest that you replace with `src/hello`